### PR TITLE
add search feature for api hub page

### PR DESCRIPTION
**Pull request by Charles-Okoeguale**

### CODE REVIEW

**Comment by Charles-Okoeguale** [comment]
Screenshot 2023-02-22 at 19 03 25

File: src/pages/Hub.tsx [hunk 1]
```diff
@@ -18,9 +21,29 @@ const Hub = () => {
 
   const [allApis, setAllApis] = useState<ApiProps[]>(apis);
   const [selectedCategoryId, setSelectedCategoryId] = useState<string>();
+  const [query, setQuery] = useState<string>('')
 
   const { categories } = useAppSelector((store) => store.apis);
 
+  const {data} = useQuery(['search for APIs'], () => 
+    axios(`${url}/api?searchBy=${query}`
```
Comment (PhilemonBrain) [inline src/pages/Hub.tsx:29]: @Charles-Okoeguale why are you not using the hook?

Comment (PhilemonBrain) [reply src/pages/Hub.tsx:29]: the fetch hook

File: src/pages/Hub.tsx [hunk 2]
```diff
@@ -18,9 +21,29 @@ const Hub = () => {
...
+    let result = data?.data.data.filter((item : ApiProps) => item.name && item.name.toLowerCase() 
+    && item.name.toLowerCase().includes(query.toLocaleLowerCase()))
+    setAllApis(result)
+  }
+
+  const debounceChange = debounce(updateQuery, 100)
+
+  useEffect(() => {
+    if (!query) {
+      setAllApis(apis)
```
Comment (PhilemonBrain) [inline src/pages/Hub.tsx:43]: what happens when a user is in a category and clears the search tab, wont this populate all APIS in that particular category?

**Review by PhilemonBrain — COMMENTED** [review]
can u respond to the issue i pointed out?

**Comment by Charles-Okoeguale** [comment]
on it

**Comment by Charles-Okoeguale** [comment]
issue resolved

**Review by PhilemonBrain — APPROVED** [review]

**Pull request merged**